lib/client/db/dbcmd/dbcmd.go

```diff
@@ -519,11 +528,27 @@ func (c *CLICommandBuilder) getMongoCommand() *exec.Cmd {
 
 	// use `mongosh` if available
 	if useMongosh {
-		return exec.Command(mongoshBin, args...)
+		// DocumentDB does not support retryable writes.
+		// https://docs.aws.amazon.com/documentdb/latest/developerguide/functional-differences.html#functional-differences.retryable-writes
+		if db.GetType() == types.DatabaseTypeDocumentDB {
+			args = append(args, "--retryWrites=false")
+		}
```

Comment:
Half of the files in this PR got changed because it needs a `types.Database` here.